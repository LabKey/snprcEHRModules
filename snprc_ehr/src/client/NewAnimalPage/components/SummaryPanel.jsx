import React from 'react'
import moment from 'moment'
import { OverlayTrigger } from 'react-bootstrap'
import SummaryPopover from '../../Shared/components/SummaryPopover'
import InfoPanel from '../../Shared/components/InfoPanel'

export default class SummaryPanel extends React.Component {
  handleKeyPress = e => {
    e.preventDefault()
  }
handlePaste = e => {
    e.preventDefault()
  }
render() {
    const { acquisitionType, acqDate, birthDate, gender, dam, sire, animalAccount,
      colony, pedigree, iacuc, ownerInstitution, responsibleInstitution, room, cage,
      diet, sourceLocation } = this.props.newAnimalData
    const { numAnimals } = this.props

    return (
      <>
        <div className="summary-panel__rows">
          <div className="section-header">Acquisition</div>
          <div className="summary-panel__row">  {/* Acquisition */}
            <div className="summary-panel__col date-width">
              <label className="summary-label">Date</label>
              <OverlayTrigger overlay={ (
                <SummaryPopover
                  message={ acqDate && moment(acqDate.date).format('MM/DD/YYYY h:mm A') }
                  title="Acquisition Date"
                />
              ) }
              >
                <input
                  className="summary-date-input"
                  defaultValue={ acqDate && moment(acqDate.date).format('MM/DD/YYYY h:mm A') }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
            <div className="summary-panel__col ">
              <label className="summary-label">Type</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ acquisitionType && acquisitionType.label } title="Acquisition Type" /> }>
                <input
                  className="summary-text-input"
                  defaultValue={ acquisitionType && acquisitionType.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                />
              </OverlayTrigger>
            </div>
            { sourceLocation
              && (
              <div className="summary-panel__col ">
                <label className="summary-label">Source Location</label>
                <OverlayTrigger overlay={ <SummaryPopover message={ sourceLocation && sourceLocation.label } title="Source Location" /> }>
                  <input
                    className="summary-text-input"
                    defaultValue={ sourceLocation && sourceLocation.label }
                    onKeyPress={ this.handleKeyPress }
                    onPasteCapture={ this.handlePaste }
                  />
                </OverlayTrigger>
              </div>
)}
            { numAnimals && numAnimals !== 1
              && (
              <div className="summary-panel__col ">
                <label className="summary-label summary-label-red"># Animals</label>
                <OverlayTrigger overlay={ <SummaryPopover message={ numAnimals && numAnimals } title="Number of animals" /> } placement="left">
                  <input
                    className="summary-text-input summary-text-input-red"
                    defaultValue={ numAnimals }
                    onKeyPress={ this.handleKeyPress }
                    onPasteCapture={ this.handlePaste }
                  />
                </OverlayTrigger>
              </div>
)}
          </div>
          <div className="section-header">Demographics</div>
          <div className="summary-panel__row"> {/* Demographics */}
            <div className="summary-panel__col date-width">
              <label className="summary-label">Birthdate</label>
              <OverlayTrigger overlay={ (
                <SummaryPopover
                  message={ birthDate && moment(birthDate.date).format('MM/DD/YYYY h:mm A') }
                  title="Birthdate"
                />
              ) }
              >
                <input
                  className="summary-date-input"
                  defaultValue={ birthDate && moment(birthDate.date).format('MM/DD/YYYY h:mm A') }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  style={ { width: '15rem' } }
                  type="text"
                />
              </OverlayTrigger>
            </div>
            <div className="summary-panel__col">
              <label className="summary-label">Gender</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ gender && gender.label } title="Gender" /> }>
                <input
                  className="summary-text-input"
                  defaultValue={ gender && gender.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
            {dam
              && (
                <div className="summary-panel__col">
                  <label className="summary-label">Dam</label>
                  <OverlayTrigger overlay={ <SummaryPopover message={ dam && dam.label } title="Dam" /> } placement="left">
                    <input
                      className="summary-text-input"
                      defaultValue={ dam && dam.label }
                      onKeyPress={ this.handleKeyPress }
                      onPasteCapture={ this.handlePaste }
                      style={ { width: '10rem' } }
                      type="text"
                    />
                  </OverlayTrigger>
                </div>
              )}
            {sire
              && (
                <div className="summary-panel__col">
                  <label className="summary-label">Sire</label>
                  <OverlayTrigger overlay={ <SummaryPopover message={ sire && sire.label } title="Sire" /> } placement="left">
                    <input
                      className="summary-text-input"
                      defaultValue={ sire && sire.label }
                      onKeyPress={ this.handleKeyPress }
                      onPasteCapture={ this.handlePaste }
                      style={ { width: '10rem' } }
                      type="text"
                    />
                  </OverlayTrigger>
                </div>
              )}
          </div>
          <div className="section-header">Location</div>
          <div className="summary-panel__row"> {/* Location */}
            <div className="summary-panel__col date-width">
              <label className="summary-label">Date</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ acqDate && moment(acqDate.date).format('MM/DD/YYYY h:mm A') } title="Move Date" /> }>
                <input
                  className="summary-date-input"
                  defaultValue={ acqDate && moment(acqDate.date).format('MM/DD/YYYY h:mm A') }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
            <div className="summary-panel__col">
              <label className="summary-label">Location</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ room && room.label } title="Location" /> }>
                <input
                  className="summary-text-input"
                  defaultValue={ room && room.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
            {cage
              && (
                <div className="summary-panel__col">
                  <label className="summary-label">Cage</label>
                  <OverlayTrigger overlay={ <SummaryPopover message={ cage && cage.value } title="Cage" /> } placement="left">
                    <input
                      className="summary-text-input"
                      defaultValue={ cage && cage.value !== undefined ? cage.value : 'N/A' }
                      onKeyPress={ this.handleKeyPress }
                      onPasteCapture={ this.handlePaste }
                      type="text"
                    />
                  </OverlayTrigger>
                </div>
              )}
          </div>
          <div className="section-header">Account, Colony, and Ownership</div>
          <div className="summary-panel__row1">   {/* Account, Colony, and Ownership */}
            <div className="summary-panel__col">
              <label className="summary-label">Account</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ animalAccount && animalAccount.label } title="Account" /> }>
                <input
                  className="summary-text-input"
                  defaultValue={ animalAccount && animalAccount.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />

              </OverlayTrigger>

            </div>
            <div className="summary-panel__col">
              <label className="summary-label">Colony</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ colony && colony.label } title="Colony" /> }>
                <input
                  className="summary-text-input"
                  defaultValue={ colony && colony.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
            <div className="summary-panel__col">
              <label className="summary-label">IACUC</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ iacuc && iacuc.label } title="IACUC" /> } placement="left">
                <input
                  className="summary-text-input"
                  defaultValue={ iacuc && iacuc.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
          </div>
          <div className="summary-panel__row">   {/* Account, Colony, and Ownership */}
            {pedigree
              && (
                <div className="summary-panel__col">
                  <label className="summary-label">Pedigree</label>
                  <OverlayTrigger overlay={ <SummaryPopover message={ pedigree && pedigree.label } title="Pedigree" /> } placement="left">
                    <input
                      className="summary-text-input"
                      defaultValue={ pedigree && pedigree.label }
                      onKeyPress={ this.handleKeyPress }
                      onPasteCapture={ this.handlePaste }
                      type="text"
                    />
                  </OverlayTrigger>
                </div>
              )}
            <div className="summary-panel__col">
              <label className="summary-label">Owner</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ ownerInstitution && ownerInstitution.label } title="Owner" /> } placement="right">
                <input
                  className="summary-text-input"
                  defaultValue={ ownerInstitution && ownerInstitution.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
            <div className="summary-panel__col">
              <label className="summary-label">Resp. Inst.</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ responsibleInstitution && responsibleInstitution.label } title="Responsible Institution" /> } placement="left">
                <input
                  className="summary-text-input"
                  defaultValue={ responsibleInstitution && responsibleInstitution.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
          </div>
          <div className="section-header">Diet</div>
          <div className="summary-panel__row"> {/* Diet */}
            <div className="summary-panel__col date-width">
              <label className="summary-label">Date</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ acqDate && moment(acqDate.date).format('MM/DD/YYYY h:mm A') } title="Diet Date" /> }>
                <input
                  className="summary-date-input "
                  defaultValue={ acqDate && moment(acqDate.date).format('MM/DD/YYYY h:mm A') }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                />
              </OverlayTrigger>
            </div>
            <div className="summary-panel__col">
              <label className="summary-label">Diet</label>
              <OverlayTrigger overlay={ <SummaryPopover message={ diet && diet.label } title="Diet" /> }>
                <input
                  className="summary-text-input"
                  defaultValue={ diet && diet.label }
                  onKeyPress={ this.handleKeyPress }
                  onPasteCapture={ this.handlePaste }
                  type="text"
                />
              </OverlayTrigger>
            </div>
          </div>
          <InfoPanel
            includeBullets
            infoMessages={ this.props.infoMessages }
          />
        </div>
      </>
    )
  }
}
