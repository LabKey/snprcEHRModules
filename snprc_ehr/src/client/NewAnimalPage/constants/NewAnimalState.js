import moment from 'moment';

const NewAnimalState = () => {

    return  ({
        currentPanel: 1,
        selectedOption: undefined,
        newAnimalData: {
            selectedOption: undefined,
            id: undefined,
            birthDate: { date: moment() },
            bdStatus: undefined,
            acquisitionType: undefined,
            acqDate: { date: moment() },
            gender: undefined,
            sire: undefined,
            dam: undefined,
            species: undefined,
            colony: undefined,
            animalAccount: undefined,
            ownerInstitution: undefined,
            responsibleInstitution: undefined,
            room: undefined,
            cage: { value: undefined },
            diet: undefined,
            pedigree: undefined,
            iacuc: undefined
        },
        summaryData: [],
        speciesList: [],
        acquisitionTypeList: [],
        potentialDamList: [],
        potentialSireList: [],
        locationList: [],
        accountList: [],
        colonyList: [],
        dietList: [],
        institutionList: [],
        iacucList: [],
        pedigreeList: [],
        bdStatusList: [],
        isLoading: true,
        hasError: false,
        preventNext: true,
        saveOk: false,
        showSaveModal: false,
        showCancelModal: false,
        showSpeciesChangeModal: false,
        errorMessage: undefined
    });
}

export default NewAnimalState;
