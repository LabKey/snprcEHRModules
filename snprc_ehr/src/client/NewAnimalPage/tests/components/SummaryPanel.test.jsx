import React from 'react'
import { shallow } from 'enzyme'
import SummaryPanel from '../../components/SummaryPanel'
import NewAnimalState from '../../constants/NewAnimalState'

describe('AcquisitionPanel tests', () => {
  test('Should render the SummaryPanel', () => {
    const { newAnimalData } = new NewAnimalState()

    const wrapper = shallow(
      <SummaryPanel
        infoMessages={ [{ key: 1, value: 'Please review data before saving.' },
        { key: 2, value: 'Hover cursor over fields for full text.' }] }
        newAnimalData={ newAnimalData }

        includeBullets={ true }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('Should render the SummaryPanel for multiple acquisitions', () => {
    const newAnimalData = {
      ...(NewAnimalState().newAnimalData),
      species: { value: 'HAM', arcSpeciesCode: 'MA' }
    }

    const wrapper = shallow(
      <SummaryPanel
        infoMessages={ [{ key: 1, value: 'Multiple animals are being addeded!' },
          { key: 2, value: 'Please review data before saving.' },
          { key: 3, value: 'Hover cursor over fields for full text.' }] }
        newAnimalData={ newAnimalData }
        numAnimals={ 2 }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
