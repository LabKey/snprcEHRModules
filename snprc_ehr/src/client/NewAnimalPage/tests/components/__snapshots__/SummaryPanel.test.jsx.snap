// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the SummaryPanel 1`] = `
<Fragment>
  <div
    className="summary-panel__rows"
  >
    <div
      className="section-header"
    >
      Acquisition
    </div>
    <div
      className="summary-panel__row"
    >
        
      <div
        className="summary-panel__col date-width"
      >
        <label
          className="summary-label"
        >
          Date
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              message="12/31/1969 6:00 PM"
              title="Acquisition Date"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-date-input"
            defaultValue="12/31/1969 6:00 PM"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col "
      >
        <label
          className="summary-label"
        >
          Type
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Acquisition Type"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
          />
        </OverlayTrigger>
      </div>
    </div>
    <div
      className="section-header"
    >
      Demographics
    </div>
    <div
      className="summary-panel__row"
    >
       
      <div
        className="summary-panel__col date-width"
      >
        <label
          className="summary-label"
        >
          Birthdate
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              message="12/31/1969 6:00 PM"
              title="Birthdate"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-date-input"
            defaultValue="12/31/1969 6:00 PM"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            style={
              Object {
                "width": "15rem",
              }
            }
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Gender
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Gender"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
    </div>
    <div
      className="section-header"
    >
      Location
    </div>
    <div
      className="summary-panel__row"
    >
       
      <div
        className="summary-panel__col date-width"
      >
        <label
          className="summary-label"
        >
          Date
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              message="12/31/1969 6:00 PM"
              title="Move Date"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-date-input"
            defaultValue="12/31/1969 6:00 PM"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Location
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Location"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Cage
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Cage"
            />
          }
          placement="left"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            defaultValue="N/A"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
    </div>
    <div
      className="section-header"
    >
      Account, Colony, and Ownership
    </div>
    <div
      className="summary-panel__row"
    >
         
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Account
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Account"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Colony
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Colony"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          IACUC
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="IACUC"
            />
          }
          placement="left"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Owner
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Owner"
            />
          }
          placement="left"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Resp. Inst.
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Responsible Institution"
            />
          }
          placement="left"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
    </div>
    <div
      className="section-header"
    >
      Diet
    </div>
    <div
      className="summary-panel__row"
    >
       
      <div
        className="summary-panel__col date-width"
      >
        <label
          className="summary-label"
        >
          Date
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              message="12/31/1969 6:00 PM"
              title="Diet Date"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-date-input "
            defaultValue="12/31/1969 6:00 PM"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
          />
        </OverlayTrigger>
      </div>
      <div
        className="summary-panel__col"
      >
        <label
          className="summary-label"
        >
          Diet
        </label>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <SummaryPopover
              title="Diet"
            />
          }
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <input
            className="summary-text-input"
            onKeyPress={[Function]}
            onPasteCapture={[Function]}
            type="text"
          />
        </OverlayTrigger>
      </div>
    </div>
    <InfoPanel
      infoMessages={
        Array [
          Object {
            "key": 1,
            "value": "Please review data before saving.",
          },
          Object {
            "key": 2,
            "value": "Hover cursor over fields for full text.",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
